src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -545,9 +545,19 @@ private void ExtractAsRegularFile(string destinationFileName)
             {
                 Access = FileAccess.Write,
                 Mode = FileMode.CreateNew,
-                Share = FileShare.None,
-                PreallocationSize = Length,
+                Share = FileShare.None
             };
+            if (OperatingSystem.IsWindows())
+            {
+                // On Windows, PreallocationSize gives a perf improvement.
```

Comment:
Why are we changing `PreallocationSize` with this change? I think this change should be specific to UnixFileMode. If there is a problem with the existing code with `PreallocationSize`, I think it would make sense to make that as a separate change.

Reply:
I'll leave out the `PreallocationSize` change and create an issue for it.

Reply:
I've created https://github.com/dotnet/runtime/issues/71753.